chore(release): v0.64.0-rhobs2

vimalk78 opened the pull request:
Prepared release `v0.64.0-rhobs2`

Commit:
git: merge rhobs-scripts

sthaha commented on .github/workflows/rhobs-release.yaml:
```diff
@@ -0,0 +1,76 @@
+
+          echo ::set-output name=tag_name::$(git describe HEAD --abbrev=0)
+
+      - name: setup rhobs module import test
+        run: |
+          cd rhobs/test/import
+          go mod edit -dropreplace github.com/rhobs/obo-prometheus-operator
+          go mod edit -dropreplace github.com/rhobs/obo-prometheus-operator/pkg/client
+          go mod edit -dropreplace github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring
+          go mod tidy
+          go test ./...
```
* Should we move this one step above (soon after the tags are created) so that the images aren't pushed? 
* How does this work if the tags aren't pushed, how does go resolve `github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring v0.64.0-rhobs2` ?

vimalk78 replied:
will merge this into previous step
if tags dont get pushed the test may not work.

sthaha commented on rhobs/test/import/go.mod:
```diff
@@ -0,0 +1,114 @@
+
+go 1.20
+
+require github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring v0.64.0-rhobs2
+
+require github.com/rhobs/obo-prometheus-operator/pkg/client v0.64.0-rhobs1 // indirect
```
Should be require - github.com/rhobs/obo-prometheus-operator instead?

vimalk78 replied:
this should have been `pkg/client v0.64.0-rhobs2` not `rhobs1`.

vimalk78 replied:
updated

sthaha commented on rhobs/test/import/import_test.go:
```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 The prometheus-operator Authors
```
Ahem .. 2023 :p And not prometheus-authors 
```suggestion
// Copyright 2023 
```

vimalk78 replied:
updated

sthaha commented on rhobs/test/import/import_test.go:
```diff
@@ -0,0 +1,28 @@
+	op "github.com/rhobs/obo-prometheus-operator/pkg/operator"
+)
+
+func TestDefalThanosImage(t *testing.T) {
```
```suggestion
func TestPackageImports(t *testing.T) {
```

vimalk78 replied:
updated

sthaha commented on rhobs/test/import/import_test.go:
```diff
@@ -0,0 +1,28 @@
+
+func TestDefalThanosImage(t *testing.T) {
+	fmt.Printf("DefaultThanosImage is %s\n", op.DefaultThanosImage)
+	fmt.Printf("PrometheusRuleName is %s\n", monv1.PrometheusRuleName)
```
maybe something from client as well.

sthaha commented:
@vimalk78 I am happy to merge this and if the release fails, prepare rhobs3 but I would encourage setting up the CI on your fork to test if things work and is able to make a release to your quay.io . 

You would have to have some hack in your rhobs-release script to point `github.com/rhobs/obo-prometheus-operator` to your fork .
And perhaps it is worth keeping that hack in that file to make the future tests easier.

vimalk78 committed:
fix(rhobs): updated release script to fix rhobs dependencies

vimalk78 committed:
chore(release): v0.64.0-rhobs2

NOTE: This commit was auto-generated by
running rhobs/make-release-commit.sh script

Signed-off-by: Vimal Kumar <vimal78@gmail.com>
